src/vtok_agent/src/agent/mngtok.rs

```diff
@@ -160,6 +205,31 @@ impl DbSource {
         serde_json::from_slice::<AcmDb>(output.stdout.as_slice()).map_err(Error::AcmDbParseError)
     }
 
+    fn fetch_s3_db(bucket_name: &str, path: &str) -> Result<AcmDb, Error> {
+        debug!("Fetching cert from s3: {}/{}", bucket_name, path);
+        let s3_url = format!(
+            "s3://{}/{}",
+            bucket_name,
+            path,
+        );
+        let output = Command::new("aws")
```

Comment:
Have you considered using the AWS Rust SDK instead of using the CLI?

Reply:
The use of the Rust SDK instead of the CLI is on my agenda, but I will implement it in a separate pull request.